web-share/share-url-invalid.https.html

```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>WebShare Test: Share with an invalid URL</title>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+  </head>
+  <body>
+    <script>
+        promise_test(t => {
+          // URL is invalid in that the URL Parser returns failure (port is too
+          // large).
+          const url = 'http://example.com:65536';
```

Comment:
> Safari doesn't seem to care that the port is too large?

Tried this on Safari 10.1.1 (not Web Share obviously, but `new URL('http://example.com:65536')`). It correctly throws a TypeError. Not sure what you're seeing, or what version.

> Maybe use a more invalid URL?

Not sure what this would be. It's actually quite hard to find an invalid URL string as @foolip pointed out. This is specified as invalid and it seems like a pretty solid one, since the data structure for URL port is an "unsigned 16-bit integer", this value can't fit in there.

> (Be careful you don't end up testing URL conformance, rather than spec conformance?)

Do you mean be careful to only test Web Share conformance, not other specs? Sure, but it's hard to do in this case because Web Share needs to fail on an invalid URL, it necessarily means we need to test with a real URL and hope the user agent follows that spec correctly.